add editorial for atcoder dp contest + small wording fixes (#1749)
cpinitiative/usaco-guide: content/4_Gold/Intro_DP.mdx

---
id: intro-dp
title: 'Introduction to DP'

> author: Michael Cao, Benjamin Qi, Neo Wang

Author: ```suggestion
author: Michael Cao, Benjamin Qi
```

		title="Errichto DP #1 - Fibonacci, iteration vs recursion"
		url="https://www.youtube.com/watch?v=YBSt1jYwVfU"
	>
		Great introduction video
	</Resource>
	<Resource
		source="Youtube"
		title="Errichto DP #2 - Coin change, double counting"
		url="https://www.youtube.com/watch?v=1mtvm2ubHCY"
	>
		Errichto DP video regarding coin change
	</Resource>
	<Resource
		source="Youtube"
		title="Errichto DP #3 - Line of Wines"
		url="https://www.youtube.com/watch?v=pwpOC1dph6U"
	>
		Errichto DP problem editorial
	</Resource>
	<Resource
		source="Youtube"
		title="WilliamFiset DP Videos"
		url="https://www.youtube.com/watch?v=_tur2nPkIKo&list=PLDV1Zeh2NRsAsbafOroUBnNV8fhZa7P4u"
	>
		Animated DP videos that pertain to interview questions
	</Resource>
</Resources>

<Info title="Pro Tip">

> Usually, it is a good idea to write a slower solution first. For example, if the

Reviewer: I prefer the original

## Classical Problems

> The next few modules provide examples of some classical problems: Dynamic

Reviewer: A colon doesn't really make sense here. Maybe use brackets instead?

Author: why not?

		url="https://atcoder.jp/contests/dp/tasks"
		starred
	>

> 		Some tasks are beyond the scope of Gold. Editorials for problems A to I can be found [here](https://nwatx.me/post/atcoderdp).

Reviewer 2: ```suggestion
		Some tasks are beyond the scope of Gold. Editorials for problems A to O can be found [here](https://nwatx.me/post/atcoderdp).
```

Reviewer 3: Why?

> 		Some tasks are beyond the scope of Gold. Editorials for problems A to O can be found [here](https://nwatx.me/post/atcoderdp).

Reviewer 4: ```suggestion
		Some tasks are beyond the scope of Gold. Editorials for problems A to U can be found [here](https://nwatx.me/post/atcoderdp).
```

Reviewer 5: lol

> 		Some tasks are beyond the scope of Gold. An editorial for problems A-I can be found [here](https://nwatx.me/post/atcoderdp). Alternative implementations for problems A - L can be found [here](https://github.com/jessechoe10/CP/tree/master/AtCoder/DP%20Contest).

Reviewer: ```suggestion
		Some tasks are beyond the scope of Gold. Editorials for problems A to I can be found [here](https://nwatx.me/post/atcoderdp). Alternative implementations for problems A to L can be found [here](https://github.com/jessechoe10/CP/tree/master/AtCoder/DP%20Contest).
```

> 		Some tasks are beyond the scope of Gold. Editorials for problems A to I can be found [here](https://nwatx.me/post/atcoderdp). Alternative implementations for problems A to L can be found [here](https://github.com/jessechoe10/CP/tree/master/AtCoder/DP%20Contest).

Author: @jessechoe10 doesn't contain J

Reviewer 3: sheesh i can add later.

Reviewer 3: I'll add more solutions later lol (it shouldn't be too hard to add them)

Reviewer 2: PRs should reflect the current state 👀

Reviewer 3: What should I change it to then?

Reviewer 2: nwatx has A-O

Reviewer 6: sussy irl
